dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcRequestHandlerMapping.java

```diff
@@ -42,9 +43,16 @@ protected boolean supportContentType(String contentType) {
 
     @Override
     protected void determineHttpMessageCodec(RequestHandler handler, URL url, HttpRequest request) {
-        HttpMessageCodec codec = CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
-        handler.setHttpMessageDecoder(codec);
-        handler.setHttpMessageEncoder(codec);
+        GrpcCompositeCodec grpcCompositeCodec =
+                (GrpcCompositeCodec) CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
+        MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
+                handler.getServiceDescriptor(), handler.getMethodName(), handler.isHasStub());
+        if (methodDescriptor != null) {
+            handler.setMethodDescriptor(methodDescriptor);
+            grpcCompositeCodec.loadPackableMethod(methodDescriptor);
```

Comment:
**Security Issue**: The method 'DescriptorUtils.findMethodDescriptor' is used without input validation, potentially allowing for injection attacks. <br> **Fix**: Validate or sanitize 'handler.getMethodName()' and 'handler.isHasStub()' before using them in 'DescriptorUtils.findMethodDescriptor'. <br> **Code Suggestion**: 
 ```
 +        String methodName = sanitize(handler.getMethodName());
 +        boolean hasStub = handler.isHasStub(); // Ensure 'hasStub' is validated if necessary
 +        MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
 +                handler.getServiceDescriptor(), methodName, hasStub);
 +        if (methodDescriptor != null) {
 ```